modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/KubernetesRuntimeProvisioningService.java

```diff
@@ -342,20 +342,207 @@ public DeploymentLogs getRuntimeLogs(DeploymentConfig deploymentConfig, Query qu
+            switch (runtimeProperty.getPropertyType()) {
+            case SENSITIVE:
+                if (log.isDebugEnabled()) {
+                    String message = "Updating property type secret for the application:" + applicationContext.getName()
+                            + " for the tenant domain:" + applicationContext.getTenantInfo().getTenantDomain();
+
+                    log.debug(message);
+                }
+                Secret secret = new SecretBuilder().withKind(KubernetesPovisioningConstants.KIND_SECRETS)
+                        .withApiVersion(Secret.ApiVersion.V_1)
+                        .withNewMetadata()
+                        .withNamespace(namespace.getMetadata().getNamespace())
+                        .withName(runtimeProperty.getName())
+                        .endMetadata()
+                        .withData(runtimeProperty.getProperties())
+                        .build();
+
+                KubernetesClient kubernetesClient = KubernetesProvisioningUtils.getFabric8KubernetesClient();
+                kubernetesClient.secrets().inNamespace(namespace.getMetadata().getName())
+                        .withName(runtimeProperty.getName()).replace(secret);
+
+                Volume volume = new VolumeBuilder()
+                        .withName(KubernetesPovisioningConstants.VOLUME_MOUNT)
+                        .withNewSecret()
+                        .withSecretName(runtimeProperty.getName())
+                        .endSecret()
+                        .build();
+
+                secrets.add(volume);
+
+                break;
+            case ENVIRONMENT:
+                if (log.isDebugEnabled()) {
+                    String message = "Updating property type environment for the application:"
+                            + applicationContext.getName() + " for the tenant domain:"
+                            + applicationContext.getTenantInfo().getTenantDomain();
+
+                    log.debug(message);
+                }
+                for (Map.Entry<String, String> property : runtimeProperty.getProperties().entrySet()) {
+                    EnvVar envVar = new EnvVarBuilder()
+                            .withName(property.getKey())
+                            .withValue(property.getValue())
+                            .build();
+
+                    environments.add(envVar);
+                }
+
+                break;
+            default:
+                String message = "Runtime property type is not support, property type:"
+                        + runtimeProperty.getPropertyType();
+
+                throw new IllegalArgumentException(message);
+            }
+        }
+
+        container.setEnvVariables(environments);
+        deploymentConfig.setContainer(container);
+        deploymentConfig.setSecrets(secrets);
+
+        //call application deployment to re deploy the application with update variables
+        deployApplication(deploymentConfig);
     }
 
+    /**
```

Comment:
This is fixed in following commit.
https://github.com/wso2/product-af/pull/800/commits/7d70ba5f5bbacfe7789db67072db49c2ff893b78